perception/object_recognition/detection/lidar_apollo_instance_segmentation/src/node.cpp

```diff
@@ -17,19 +17,21 @@
 #include "lidar_apollo_instance_segmentation/node.h"
 #include "lidar_apollo_instance_segmentation/detector.h"
 
-LidarInstanceSegmentationNode::LidarInstanceSegmentationNode() : nh_(""), pnh_("~")
+LidarInstanceSegmentationNode::LidarInstanceSegmentationNode() : Node("lidar_apollo_instance_segmentation_node")
 {
-  detector_ptr_ = std::make_shared<LidarApolloInstanceSegmentation>();
+  using std::placeholders::_1;
+  detector_ptr_ = std::make_shared<LidarApolloInstanceSegmentation>(this);
+  debugger_ptr_ = std::make_shared<Debugger>(this);
   pointcloud_sub_ =
-    pnh_.subscribe("input/pointcloud", 1, &LidarInstanceSegmentationNode::pointCloudCallback, this);
-  dynamic_objects_pub_ = pnh_.advertise<autoware_perception_msgs::DynamicObjectWithFeatureArray>(
+    this->create_subscription<sensor_msgs::msg::PointCloud2>("input/pointcloud", 1, std::bind(&LidarInstanceSegmentationNode::pointCloudCallback, this, _1));
+  dynamic_objects_pub_ = this->create_publisher<autoware_perception_msgs::msg::DynamicObjectWithFeatureArray>(
     "output/labeled_clusters", 1);
 }
 
-void LidarInstanceSegmentationNode::pointCloudCallback(const sensor_msgs::PointCloud2 & msg)
+void LidarInstanceSegmentationNode::pointCloudCallback(const sensor_msgs::msg::PointCloud2::ConstSharedPtr msg)
 {
-  autoware_perception_msgs::DynamicObjectWithFeatureArray output_msg;
-  detector_ptr_->detectDynamicObjects(msg, output_msg);
-  dynamic_objects_pub_.publish(output_msg);
-  debugger_.publishColoredPointCloud(output_msg);
+  autoware_perception_msgs::msg::DynamicObjectWithFeatureArray output_msg;
+  detector_ptr_->detectDynamicObjects(*msg, output_msg);
+  dynamic_objects_pub_->publish(output_msg);
+  // debugger_ptr_->publishColoredPointCloud(output_msg);
```

Comment:
Is this commented out because it is not used anywhere?

Reply:
Fixed.